ui/core/sim.ts

```diff
@@ -243,6 +245,81 @@ export class Sim {
 				if (gearChanged) {
 					player.equipment = gear.asSpec();
 				}
+
+				// Include consumables in the player db
+				const pdb = player.database!;
+				pdb.armorDb = this.db.getArmorDb();
+				pdb.weaponDamageDb = this.db.getWeaponDamageDb();
+				pdb.armorTotalValue = [] as ItemArmorTotal[];
+				pdb.randomPropPoints = [] as QualityAllocations[];
+				gear.asArray().forEach(gear => {
+					if (!pdb.randomPropPoints[gear?.item.ilvl ?? 0]) {
+						if (this.db.getQualityAllocByIlvl(gear?.item.ilvl ?? 0)) {
+							pdb.randomPropPoints[gear?.item.ilvl ?? 0] = this.db.getQualityAllocByIlvl(gear?.item.ilvl ?? 0)!;
+						}
+					}
+					if (!pdb.armorTotalValue[gear?.item.ilvl ?? 0]) {
+						if (this.db.getArmorTotalValueByIlvl(gear?.item.ilvl ?? 0)) {
```

Comment:
Maybe cache this value `this.db.getArmorTotalValueByIlvl(gear?.item.ilvl ?? 0)` so when it passes the if you don't have to do it again, but rather you can just assign it?